### ISSUE TITLE
StaticQuery preview support

### ISSUE THREAD

**Issue by ffdead** [birkir/gatsby-source-prismic-graphql#9]
Hi and thanks again for all the work with this library!

Is it possible to enable preview on a `StaticQuery` somehow? I tried wrapping it in `withPreview()` manually but it didn't seem to work.

Thanks!

**Comment by ffdead** [comment]
I worked around this issue like this for now: 

```js
const query = graphql`
  {
    prismic {
      site_settings(uid: "site-settings", lang: "en-gb") {
        ...
      }
    }
  }
`
export default withPreview(({data: previewData, ...props}) => {
  if (previewData) {
    return <Layout {...props} prismic={previewData.prismic} />;
  }
  return (
    <StaticQuery
      query={query}
      render={data => <Layout {...props} prismic={data.prismic} />}
    />
  )
}, query)

```

If we can find a solution for #10, I think we are in a good place! 🤞

**Comment by ffdead** [comment]
This workaround no longer works with `3.0.0-alpha.14` since the removal of `withPreview`.

**Comment by birkir** [comment]
Fixed with 3.0.0-alpha.21

**Comment by mvogel-dev** [comment]
Has this bug popped up again? I've tried multiple variations of withPreview with StaticQuery from the various examples and the main readme documentation without any luck. It pulls the original data at build time, but previewing doesn't seem to be working on components using StaticQuery, even with the withPreview wrapper.

Version 3.1.3
